pkg/manifests/bundleloader.go

```diff
@@ -49,33 +49,58 @@ func (b *bundleLoader) LoadBundle() error {
 
 // Compress the bundle to check its size
 func (b *bundleLoader) calculateCompressedBundleSize() error {
+	if b.bundle == nil {
+		return nil
+	}
 	err := filepath.Walk(b.dir,
 		func(path string, info os.FileInfo, err error) error {
 			if err != nil {
 				return err
 			}
+			if info.IsDir() {
+				return nil
+			}
+			data, err := os.ReadFile(path)
+			if err == nil {
+				fileSize, err := fileSize(path)
+				if err != nil {
+					return err
+				}
+				contentGzip, err := encoding.GzipBase64Encode(data)
```

Comment:
Is the bundle compression really to Base64 or is it raw? The util method for gzip+base64 used here results in a human-readable, but potentially larger, compressed file. I see this allows us to check with `len` later which is convenient, but it's predicated on the base64 yielding text.

I will search for the code for the bundle compression and add a link here if I find it.

Reply:
It is the same code. used by OLM
So, the idea here is we keep the same logic to check that.

Reply:
I wasn't able to find in OLM repo searching on gzip...

Reply:
See; https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/encoding/encoding.go
It was moved to the API.
The next. step is to remove from OPM the implementation and import it from here.

Reply:
Marking as sorted out.

Reply:
:sgtm